Climaxing/masturbation changes, exposing through clothing, breastmilk and masturbation

ktccd opened the pull request:
[Changelogs]: # (Please make a changelog if you're adding, removing or changing content that'll affect players. This includes, but is not limited to, new features, sprites, sounds; balance changes; map edits and important fixes)

:cl: ktccd
add: Breasts can now be used to masturbate/milk yourselves.
add: You can now expose genitals through clothing using a verb in the IC tab.
tweak: Pressing the Arousal HUD symbol will now bring up several options to use. Pull or be pulled by someone who's exposed to have them appear as an option.
tweak: Made hexacamphor/hexacrocin less verbose and trigger forced climaxes less often to avoid incredible spam.
tweak: Being forced to orgasm while pulled or pulling someone now counts them as your selected partner.
/:cl:

[why]: # (Please add a short description [on the next line] of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding:) 
Lots of things:
Exposing genitalia has been asked for a while now, and helps work around things were normal jackets would count as covering the groin.
Hexacrocin and hexacamphor were too verbose and the forced orgasms happening too often.
Breastcode existed to support it, but there was no way for players to milk themselves. This is now supported.
My previous rework of the masturbation was a bit lacking in options and the bloated look of the proc really added to that.
You can now also cum on any food item, but I'm not putting that in the changelogs.... I feel a lot of people would break the OOC-consent rule by feeding people their semen, without thinking about needing consent....


~~DO NOT MERGE JUST YET, I gotta clean up some comments I think I forgot?~~
Ok, removed the comments.

ktccd committed:
Genital refactoring and masturbation rework

Refactors a lot of genitalia code.
Reworks masturbation code.
Adds the ability to expose onself through clothing (Still bugfixing it).

ktccd committed:
Fixes it so everything works

Toggling exposing your genitals now works properly. You can now expose
genitals through clothing.
Genitalia is rendered a bit more modularly and less hardcoded.
Genitalia no longer has linked_penis and such, but instead linked_organ.
More variables made general for genitals.
Breasts can now be used to masturbate and climax, also letting you fill
beakers and others with milk.
Both being pulled and pulling someone can let you count them as a
partner for climaxing verbs.
Mob_climax split up into more sub-procs.
Hexacrocin made less verbose, and not as spam-happy with forced orgasms.
Hexacamphor also made less verbose.

ktccd committed:
Removes outcommented code

Forgot to remove the old masturbation code that I commented out.

deathride58 commented on code/citadel/cit_arousal.dm:
```diff
@@ -181,224 +181,365 @@
+			src.visible_message("<span class='danger'>[src] orgasms[istype(src.loc, /turf/open/floor) ? ", spilling onto [src.loc]" : ""], using [p_their()] [G]!</span>", \
+								"<span class='green'>You climax[istype(src.loc, /turf/open/floor) ? ", spilling onto [src.loc]" : ""] with your [G].</span>", \
+								"<span class='green'>You climax using your [G].</span>")
+		if(G.can_climax)
```
You forgot a tab here. Without that tab, your arousal will reset even if the timer gets interrupted.

ktccd replied:
ah, dang tabs! Good eye ^^.

ktccd replied:
Fixed it.

ktccd committed:
Changes one tab further on line 248

As pointed out to me ^^.

deathride58 commented on code/citadel/cit_arousal.dm:
```diff
@@ -181,224 +181,365 @@
+			src.visible_message("<span class='danger'>[src] climaxes with [L], [p_their()] [G] spilling nothing!</span>", \
+								"<span class='green'>You ejaculate with [L], your [G] spilling nothing.</span>", \
+								"<span class='green'>You have climaxed inside someone, your [G] spilling nothing.</span>")
+	if(G.can_climax)
```
bad tabs 2: electric boogaloo
if you get separated from your partner or you move, your arousal will reset, but you will not cum. slightly realistic, but probably not the intended effect.

ktccd replied:
Fixed. I added the check to the spillage also. I think my idea was for the command to run after either the spillage and non-spillage versions, but I forgot about the do_after part being important for both.

ktccd committed:
Climaxing with partner tabulation

Also copied the command for the spillage version.

ktccd committed:
capitalisation and timers

Removed timers from forced climaxes and made it a default value
parameter instead, so it's adjustable.
Changed capitalisation of genitals and the mentions of them now use the
.name to avoid including a "The" in the mention.

ktccd committed:
I accidentally runtimestation

Accidentally committed changing the map to runtimestation when I
debugged my code.
Changed it back to basemap now.

TalkingCactus commented on code/citadel/cit_reagents.dm:
```diff
@@ -153,14 +153,13 @@
 			var/mob/living/carbon/human/H = M
-			to_chat(H, "<span class='love'>Your libido is going haywire!</span>")
-			H.mob_climax(forced_climax=TRUE)
+			if(prob(50)) //Less spam
```
I'd go 10 here.

ktccd replied:
I tested that and it almost never triggered.
It's prob(33) THEN prob(50), resulting in about a prob(16) total. If I made it prob(10), the end result would be a 3% chance to trigger :S.

TalkingCactus replied:
It doesn't need to tell you that often, really. >20u of the stuff lasts a while anyway, so you'll still get the message a fair few times while it metabolizes.

ktccd replied:
yeah... but 3% is extremely low. I just remember testing that and I basically never saw a message or was forced to climax...

ktccd committed:
Merge remote-tracking branch 'origin/Genitalia-tweaks' into Genitalia-tweaks

# Conflicts:
#	code/citadel/cit_arousal.dm
Somehow I got conflicts with my own damn commits.... What... How does
that even work....
Anyway, I think I fixed it with this.
